### PR TITLE
editoast: add telemetry on database requests

### PR DESCRIPTION
**Pull request by woshilapin**
The second commit tries to add identifiers in metadata of the traces. However, it seems that it would be not trivial to do it without the `valuable` feature... which is unstable.

⚠️ Weirdly, the build with Docker seems to ignore the configuration `.cargo/config.toml` (hence the build is failing). Needs some investigation if we want to go that path.

### CODE REVIEW

**Comment by leovalais** [comment]
Nice! I'm not against using the unstable feature as even if that becomes a problem, reverting that single commit wouldn't be much work and won't degrade the service. But I'm wondering whether we should be using `Valuable` at all as `Debug` seems to be fine (albeit less readable). And even if we were to parse the span metadata with some script in the future, we can always deserialize it as RON.

Maybe `model-X` would be a better span name than `database-X` (an object may require several model operations to be considered "retrieved" from the database, eg. scenarios that also fetch their study and project).

Now that `List` and `Count` are merged they also need to be annotated ;)

**Comment by woshilapin** [comment]
I removed the unstable feature and went `Debug` everywhere. Looks much better now.

File: editoast/.cargo/config.toml
Comment (leovalais) [inline editoast/.cargo/config.toml:1]: Is this file still necessary?

**Review by flomonster — APPROVED** [review]
LGTM! (tested + reviewed)

File: editoast/editoast_derive/src/modelv2/codegen/count_impl.rs [hunk 1]
```diff
@@ -9,11 +9,19 @@ pub(crate) struct CountImpl {
...
         let Self { model, table_mod } = self;
+        let span_name = format!("model:list<{}>", model);
 
         tokens.extend(quote! {
             #[automatically_derived]
             #[async_trait::async_trait]
             impl crate::modelsv2::prelude::Count for #model {
+                #[tracing::instrument(name = #span_name, skip_all, ret, err, fields(
+                    nb_filters = settings.filters.len(),
+                    nb_sorts = settings.sorts.len(),
```
Comment (leovalais) [inline editoast/editoast_derive/src/modelv2/codegen/count_impl.rs]: Sorting is irrelevant for counting, but some sorts might be there if the settings are used elsewhere, you can remove it if you want but I'm fine leaving it like this.

File: editoast/src/modelsv2/mod.rs [hunk 2]
```diff
@@ -157,7 +157,7 @@ mod tests {
             }
         }
 
-        #[derive(Clone, ModelV2)]
```
Comment (leovalais) [inline editoast/src/modelsv2/mod.rs:160]: You may also add this bound to `modelsv2::trait Model`

**Review by leovalais — APPROVED** [review]
Looks great! Since @flomonster already tested it I'll just play with it later ;)

File: editoast/.cargo/config.toml
Comment (woshilapin) [reply editoast/.cargo/config.toml:1]: Nope, I removed it 🤔

File: editoast/src/modelsv2/mod.rs (hunk 2 above)
Comment (woshilapin) [reply editoast/src/modelsv2/mod.rs:160]: Done in ff1e7dd06.

File: editoast/editoast_derive/src/modelv2/codegen/count_impl.rs (hunk 1 above)
Comment (woshilapin) [reply editoast/editoast_derive/src/modelv2/codegen/count_impl.rs]: Removed.
